File: api/posts.js
const { kv } = require('@vercel/kv');

module.exports = async (req, res) => {
  try {
    let posts = await kv.get('posts');
    if (!posts) {
      posts = [];
    }
    
    const sortedPosts = posts.sort((a, b) => new Date(b.time) - new Date(a.time));
    
    res.status(200).json({
      all_posts: sortedPosts,
      history: sortedPosts.slice(0, 10),
      stats: {
        total: sortedPosts.length,
        generated: sortedPosts.filter(p => p.generated).length
      }
    });
  } catch (error) {
    console.error('Error in posts API:', error);
    res.status(500).json({ error: error.message });
  }
};

File: api/posts.py
import os
import json
from datetime import datetime

POSTS_FILE = 'posts.json'

def load_posts():
    if os.path.exists(POSTS_FILE):
        with open(POSTS_FILE, 'r') as f:
            return json.load(f)
    return []

def handler(request):
    posts = load_posts()
    sorted_posts = sorted(posts, key=lambda x: datetime.fromisoformat(x['time']), reverse=True)
    
    return {
        'statusCode': 200,
        'headers': {'Content-Type': 'application/json'},
        'body': json.dumps({
            'all_posts': sorted_posts,
            'history': sorted_posts[:10],
            'stats': {
                'total': len(sorted_posts),
                'generated': len([p for p in sorted_posts if p.get('generated')])
            }
        })
    }
